lib/BloqadeQMC/test/ltfim.jl

```diff
@@ -94,41 +96,103 @@ expected_values = Dict{Tuple{Bool, Int, Float64}, Dict{String, Float64}}(
 #     @test stdscore(energy, expected_vals["H"]) < THRESHOLD
 # end
 
+THRESHOLD = 2.576  # 99% Two-sided CI of the t-distribution with infinite dofs
 
-@testset "1D LTFIM Thermal State $N-sites, PBC=$PBC, β=1.0" for PBC in [true, false], N in [5, 10]
+@testset "1D LTFIM Thermal State $N-sites, PBC=$PBC, β=1.0" for PBC in [true], N in [5]
     # rng = Xorshifts.Xoroshiro128Plus(1234)
-    rng = MersenneTwister(4321)
+    rng = MersenneTwister(1432)
 
-    H = LTFIM((N,), 1.0, 1.0, 1.0, PBC)
+    H = LTFIM((N,), 1.0, 0.0, 0.0, PBC)
     th = BinaryThermalState(H, 1000)
     beta = 1.0
+    d = Diagnostics()
 
     MCS = 1_000_000
     EQ_MCS = 100_000
 
     mags = zeros(MCS)
     ns = zeros(MCS)
 
-    [mc_step_beta!(rng, th, H, beta) for i in 1:EQ_MCS]
+    [mc_step_beta!(rng, th, H, beta, d) for i in 1:EQ_MCS]
 
     for i in 1:MCS # Monte Carlo Steps
-        ns[i] = mc_step_beta!(rng, th, H, beta) do lsize, th, H
-            mags[i] = magnetization(sample(H, th))
+        ns[i] = mc_step_beta!(rng, th, H, beta, d) do lsize, th, H
+            mags[i] = magnetization(sample(H, th, 1))
         end
     end
+
+    # Binning Analysis for observables based on mags
+    B = LogBinner(abs.(mags))
+    println("Correlation time for abs_mag_binned samples")
+    τ_abs = tau(B)
+    println(τ_abs)
+    println("N_eff")
+    N_eff = MCS / τ_abs
```

Comment:
LogBinner uses a slightly different convention for tau, so N_eff should be `MCS / (2*tau + 1)`